book/src/kernel/linux-compatibility/limitations-on-system-calls/system-call-matching-language.md

```diff
@@ -176,6 +176,53 @@ poll(fds = [ <pollfd> ], nfds, timeout);
 
 Notice how SCML denotes an array with the `[ <struct_rule> ]` syntax.
 
+SCML supports arrays with heterogeneous element types,
+as encountered in system calls like `recvmsg` where netlink message payloads
+follow the TLV (Type-Length-Value) format.
+In such cases, array elements can have varying structures or types,
+and SCML is designed to match these patterns as they appear in strace output.
+
+When an array contains multiple element types, they should be separated by commas.
+For example, when receiving a netlink message about adding a network address:
+
+```c
+struct msg_iov = {
```

Comment:
I think you meant `struct iovec`?

```c
struct iovec
{
	void* iov_base;
	size_t iov_len;
};
```

The field name is `msg_iov`, whose type name is `struct iovec`. See Linux's source code: https://elixir.bootlin.com/linux/v6.17.3/source/include/linux/socket.h#L83-L91